### PR TITLE
Fix clippy warnings&erros

### PR DESCRIPTION
**Pull request by Aloxaf**
This is a very simple PR: I ran clippy and corrected the code as it wants.

[Clippy](https://github.com/rust-lang/rust-clippy) is a good tool to improve your Rust code :D

### CODE REVIEW

**Comment by gahag** [comment]
Thanks for taking your time into this! Clippy seems to be great, thanks for telling me about it. I believe it found one real bug in my code, and your PR solves it. I just have some minor reviews.

File: src/grep.rs [hunk 1]
```diff
@@ -11,7 +11,7 @@ use crate::args::{self, Args};
 /// Build the regex pattern with the given options.
 /// By default, the `unicode` flag is set to false, and `dot_matches_new_line` set to true.
 fn build_pattern(
-  pattern: &String,
+  pattern: &str,
```
Comment (gahag) [inline src/grep.rs]: This part is already adressed in #1 , so I'll have it from there. Can you revert this part please?

File: src/grep.rs [hunk 2]
```diff
@@ -89,7 +89,7 @@ fn grep_bytes(
       write!(stdout, "{}: ", path)?;
     }
 
-    stdout.write(bs)?;
+    stdout.write_all(bs)?;
```
Comment (gahag) [inline src/grep.rs:92]: I believe this solves a bug. A call to `write` that doesn't output all bytes would be considered successful. Many thanks for finding this.

**Review by gahag — CHANGES_REQUESTED** [review]

**Comment by Aloxaf** [comment]
Thanks for your advice. I have reverted it by a new commit. Is this OK?

**Comment by gahag** [comment]
This is perfect. Once again, many thanks for your contribution. I will merge this tomorrow as soon as I get on my pc.

**Review by gahag — APPROVED** [review]

**Pull request merged**